File: app/src/main/java/com/falcon/restaurants/screens/common/dialogs/BaseDialog.java
package com.falcon.restaurants.screens.common.dialogs;

import androidx.fragment.app.DialogFragment;
import com.falcon.restaurants.common.di.presentation.PresentationComponent;
import com.falcon.restaurants.common.di.presentation.PresentationModule;
import com.falcon.restaurants.screens.common.BaseActivity;

public class BaseDialog extends DialogFragment {

    private PresentationComponent presentationComponent;
    public PresentationComponent getPresentationComponent(){
        if(presentationComponent == null){
            presentationComponent =
                    ((BaseActivity)requireActivity()).getActivityComponent()
                            .newPresentationComponent(new PresentationModule());
        }
        return presentationComponent;
    }
}

File: app/src/main/java/com/falcon/restaurants/screens/common/dialogs/BaseDialog.kt
package com.falcon.restaurants.screens.common.dialogs;

import androidx.fragment.app.DialogFragment;
import com.falcon.restaurants.common.di.presentation.PresentationComponent;
import com.falcon.restaurants.common.di.presentation.PresentationModule;
import com.falcon.restaurants.screens.common.BaseActivity;

open class BaseDialog : DialogFragment() {

    val presentationComponent by lazy {
        (requireActivity() as BaseActivity).activityComponent
            .newPresentationComponent(PresentationModule())
    }

}
